packages/mui-base/src/useSelect/selectReducer.ts

```diff
@@ -101,6 +108,10 @@ export function selectReducer<OptionValue>(
       break;
 
     case ListActionTypes.blur:
+      if (action.context.focusManagement === 'DOM') {
+        return state;
+      }
```

Comment:
This bypass is inspired by [this other one](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/useList/listReducer.ts#L333) in the listReducer. (Note: When I mention bypass, I'm referring to the `if` statement that returns the state unchanged in lines 111-113).

There are three options to close the listbox when it loses focus:

1. What I implemented here: Add a `closeListbox` action to dispatch manually from Material UI's side and bypass the blur event with this bypass here (otherwise, the `open: false` state is set twice)
2. Add a `closeListbox` action for the user to dispatch manually and prevent the double `open: false` double setting by adding `defaultMuiPrevented = true` to the blur handler on Material UI's side.
3. Rely on the blur event to close the listbox

Option 1. is easier for Material UI: [This test](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Select/Select.test.js#L193-L211) would work without changes. In Option 3, we would have to change the test as `getByTestId('backdrop').click();` doesn't trigger a blur event. Refactoring to `userEvent.click` works as that triggers the blur event on the listbox.

Option 3. feels like the one I would go to if I weren't thinking of Material UI.

I'm having trouble figuring out what would be the best option to go for.

Reply:
I'd say Option 3 is the most sensible. Using `userEvent` in these tests does make sense as we don't care about the precise event that is fired but rather want the element to lose focus.

Reply:
Went for option 3.